src/parachain/escrow.ts

```diff
@@ -210,29 +210,38 @@ export class DefaultEscrowAPI implements EscrowAPI {
         amount: MonetaryAmount<GovernanceCurrency>;
         apy: Big;
     }> {
-        const [userStake, totalStake, blockReward, stakedBalance, currentBlockNumber, minimumBlockPeriod, maxPeriod] =
-            await Promise.all([
-                this.getEscrowStake(accountId),
-                this.getEscrowTotalStake(),
-                this.getRewardPerBlock(),
-                this.getStakedBalance(accountId),
-                this.systemAPI.getCurrentBlockNumber(),
-                this.api.consts.timestamp.minimumPeriod,
-                this.getMaxPeriod(),
-            ]);
-
-        return estimateReward(
-            this.governanceCurrency,
-            userStake,
-            totalStake,
-            blockReward,
-            stakedBalance,
-            currentBlockNumber,
-            minimumBlockPeriod.toNumber(),
-            maxPeriod.toNumber(),
-            amountToLock,
-            blockLockTimeExtension
+        const stakedBalance = await this.getStakedBalance(accountId);
+
+        let atomicStakedAmount: Big = stakedBalance.amount.toBig(ATOMIC_UNIT);
+
+        let nullableAmountToCheck: bigint | null = null;
+        if (amountToLock && !amountToLock.isZero()) {
+            atomicStakedAmount = atomicStakedAmount.add(amountToLock.toBig(ATOMIC_UNIT));
+            // get as bigint for the rpc api
+            nullableAmountToCheck = BigInt(atomicStakedAmount.toString());
+        }
+
+        let nullableBlockToEnd: bigint | null = null;
+        if (blockLockTimeExtension) {
+            const currentEndBlock = !stakedBalance.amount.isZero() ? BigInt(stakedBalance.endBlock) : BigInt(0);
+            nullableBlockToEnd = currentEndBlock + BigInt(blockLockTimeExtension);
+        }
+
+        const rawRewardRate = await this.api.rpc.reward.estimateEscrowRewardRate(
+            accountId,
+            nullableAmountToCheck,
+            nullableBlockToEnd
         );
+
+        // annual reward rate
+        const rewardRate = decodeFixedPointType(rawRewardRate);
+        // reward amount
+        const rewardAmount = newMonetaryAmount(atomicStakedAmount, this.governanceCurrency).mul(rewardRate);
```

Comment:
Ah I didn't realize you also needed the concrete reward amount, so this is actually wrong because in the parachain `rate = received / total_locked` where `total_locked` is `stakedBalance + amount`.

Reply:
So to get the reward amount it would be `received = rate * (stakedBalance + amount)`

Reply:
> So to get the reward amount it would be received = rate * (stakedBalance + amount)

I think I got it covered here?
https://github.com/interlay/interbtc-api/pull/538/files/592f0346f2cf87b19a23377460e019580f056cc0#diff-4573174c08a415f939d2d03353df39634595439f0f5a301e7c4f2ee7ab58d55cR219

Reply:
Ah right I missed that